_sips/minutes/2020-03-12-minutes.md

```diff
@@ -0,0 +1,166 @@
+
+### Extension methods
+
+* Gui: one difference between infix extension methods and grouped extension methods is that grouped they live in a given instance
+* Martin: people like the `extension` keyword, when we have it as only `given` people were confused by what these things were
+* Gui: one big difference with the status quo (Scala 2) of extension methods is you can't have two sets of type parameters, on the class and on the def, and people aren't pleased with that
+* Martin: eventually that will go when we have curried type parameters, which isn't a 3.0 thing
+* Gui: could we add `private` support to extension methods?
+* Martin: Actually I just tried it, we do allow private
+
+### Export clauses
+
+* Martin: addressing the problem that Scala don't assist encouraging composition (over inheritance)
+* Martin: side reasons are package object inhertance and exporting enum cases
+* Martin: the feedback is some want it to do more, some want it to do less
+* Martin: I feel good about how it is now, I wouldn't change any detail on it
+* Adriaan: it would be good to add a type ascription when exporting (`export A.T: Foo` to only export Foo's methods)
+* Martin: that wouldn't be big, we could add that
+* Gui: we could add refinements to the import, to select exactly the method used (`import A.T { def foo(x: Int): Int }`)
+* Martin: or users could just do it with a val `val x: Any { def a(x: Int): Int }`
+* Adriaan: I'm symapthetic to only allowing to export from static paths, we can always extend it to non static paths
+* Seb: the question is does the restriction buy us anything
+* Martin: importantly the forwarders are final, you can't overriden them
+* Gui: In TASTy should adding a method be TASTy compatible?
+* Seb: if we consider them implementation details, then no, you defined your API once, adding a method to the exported thing shouldn't propagate to where you previously exported it
+* Gui: export also "interacts" with visibility
+* Martin: so you can do `private export foo._`, but in order to you need access to foo at the definition point of the export
+* Gui: but now it's `export foo._` which includes all the package private things in `foo`
+* Dale: there's a precedent there with current (term and type) forwarders, but with this you can do lots at once, which aggravates the problem
+
+### Review meta programming
+
+(**Note: most of the time was spent going over the various feature changes, through the docs.**)
+
+Run through https://dotty.epfl.ch/docs/reference/metaprogramming/inline.html
+
+* Gui: why `inline msg: String`?  Seeing as the semantics match just use `msg: => String`, just as the syntax, with the current codegen implementation.
+* Gui: inline interacts with overriding
+* Nic: if in a subclass (say Range) you override a method (say foreach) with `inline` it only inlines if the type is cast
+* Seb: I need both, inline if statically the type is Range, and use the optimised override if it's a Seq that's a Range at runtime
```

Comment:
This eventually led to https://github.com/lampepfl/dotty/pull/8543 and https://github.com/lampepfl/dotty/issues/8564. I suggest pointing at least to the former as it contains concrete results of the discussion that happened during the SIP meeting.